tensorflow-framework/src/main/java/org/tensorflow/framework/activations/Activation.java

```diff
@@ -14,24 +14,104 @@
 =======================================================================*/
 package org.tensorflow.framework.activations;
 
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Supplier;
 import org.tensorflow.Operand;
 import org.tensorflow.op.Ops;
 import org.tensorflow.types.family.TNumber;
 
-/**
- * Interface for Activations
- *
- * @param <T> the data type of the input and the result
- */
+/** Interface for Activations */
 @FunctionalInterface
-public interface Activation<T extends TNumber> {
+public interface Activation {
+  Map<String, Supplier<Activation>> createMap =
+      new HashMap<String, Supplier<Activation>>() {
+        {
+          put("elu", ELU::new);
+          put("exponential", Exponential::new);
+          put("gelu", GELU::new);
+          put("hard_sigmoid", HardSigmoid::new);
+          put("linear", Linear::new);
+          put("relu", ReLU::new);
+          put("selu", SELU::new);
+          put("sigmoid", Sigmoid::new);
+          put("softmax", Softmax::new);
+          put("softplus", Softplus::new);
+          put("softsign", Softsign::new);
+          put("swish", Swish::new);
+          put("tanh", Tanh::new);
+        }
+      };
+
+  Map<String, Function<Map<String, Object>, Activation>> createMapConfig =
+      new HashMap<String, Function<Map<String, Object>, Activation>>() {
+        {
+          put("elu", ELU::new);
+          put("exponential", Exponential::new);
+          put("gelu", GELU::new);
+          put("hard_sigmoid", HardSigmoid::new);
+          put("linear", Linear::new);
+          put("relu", ReLU::new);
+          put("selu", SELU::new);
+          put("sigmoid", Sigmoid::new);
+          put("softmax", Softmax::new);
+          put("softplus", Softplus::new);
+          put("softsign", Softsign::new);
+          put("swish", Swish::new);
+          put("tanh", Tanh::new);
+        }
+      };
+
+  /**
+   * Creates an Activation based on a name as known to the TensorFlow engine.
+   *
+   * @param name the activation name
+   * @return the Activation
+   */
+  static Activation create(String name) {
+    if (name == null) {
+      return new Linear();
+    }
+    if (createMap.containsKey(name)) {
+      return createMap.get(name).get();
+    } else {
+      throw new IllegalArgumentException("Unknown conversion for Activation " + name);
+    }
+  }
+
+  /**
+   * Creates an Activation based on a configuration as produced by TensorFLow.
+   *
+   * @param config the constraint configuration, the config format is
+   *     <pre>{@code
+   * "name" : String - this is used to locate the class, this is the TensorFlow Engine's class name
+   *
+   * }</pre>
+   *
+   * @return the Activation
+   */
+  static Activation create(Map<String, Object> config) {
+    String className = config == null ? null : (String) config.get("name");
+    if (className == null) {
```

Comment:
Is there a reason we want this to default to linear rather than throwing IAE for both null and unknown names?

Reply:
New code throws NPE if null, else throws IAE if name is not a known activation